pkg/errors/type.go

```diff
@@ -13,3 +13,19 @@ var (
 
 // Defines custom error types
 type typ struct{ s string }
+
+func NotFoundWrap(cause error, code Code, format string, args ...interface{}) *base {
+	return Wrapf(cause, TypeNotFound, code, format, args...)
+}
+
+func NotFoundNew(code Code, format string, args ...interface{}) *base {
+	return Newf(TypeNotFound, code, format, args...)
+}
+
+func InternalWrap(cause error, code Code, format string, args ...interface{}) *base {
+	return Wrapf(cause, TypeInternal, code, format, args...)
+}
+
+func InternalNew(code Code, format string, args ...interface{}) *base {
+	return Newf(TypeInternal, code, format, args...)
+}
```

Comment:
```suggestion
func WrapNotFoundf(cause error, code Code, format string, args ...interface{}) *base {
	return Wrapf(cause, TypeNotFound, code, format, args...)
}

func NewNotFoundf(code Code, format string, args ...interface{}) *base {
	return Newf(TypeNotFound, code, format, args...)
}

func WrapInternalf(cause error, code Code, format string, args ...interface{}) *base {
	return Wrapf(cause, TypeInternal, code, format, args...)
}

func NewInternalf(code Code, format string, args ...interface{}) *base {
	return Newf(TypeInternal, code, format, args...)
}
```
1. Please move them to `errors.go` file. `type.go` is just meant for declaring the typ struct.
2. When naming functions, try to say them out loud. `NewNotFound` v/s `NotFoundNew`.
3. The `f` after all functions indicates that it takes some formatting input. Similar to `Printf`, `Errorf` etc.

Reply:
@grandwizard28 I have implemented the changes mentioned except point 2.

Slack Discussion reference: https://signoz-team.slack.com/archives/C089D1B5516/p1747107589304479

If we're still not aligned on this, I would request to have an open discussion around this, if not I'll implement or revert the current changes later in a separate PR. 

Would request you for an approval on this PR, as this feature is 1 week due on it's initial release.